sqla_vertica_python/vertica_python.py

```diff
@@ -200,15 +200,54 @@ def get_columns(self, connection, table_name, schema=None, **kw):
         where table_name = '{table_name}'
         {schema_conditional}
         """.format(table_name=table_name, schema_conditional=schema_conditional)
-        return [
-            {
-                'name': row.column_name,
-                'type': self.ischema_names[row.data_type.upper().split('(')[0]],
-                'nullable': row.is_nullable,
-                'default': row.column_default,
-                'primary_key': row.column_name in primary_key_columns
-            } for row in connection.execute(column_select)
+        colobjs = [ 
+            self._get_column_info(
+                row.column_name, 
+                row.data_type, 
+                row.is_nullable,
+                row.column_default,
+                (row.column_name in primary_key_columns)
+                )
+            for row in connection.execute(column_select)
         ]
+        return [ c for c in colobjs if c ]
+
+    def _get_column_info(self, name, data_type, is_nullable, default, is_primary_key):
+        m = re.match(r'(\w+)(?:\((\d+)(?:,(\d+))?\))?', data_type)
+        if not m:
+            return {}
+        typename = m.group(1).upper()
+        typeobj = self.ischema_names[typename]
+        typeargs = []
+        typekwargs = {}
+        if m.group(2):
+            try:
+                v = int(m.group(2))
+                typeargs.append(v)
+            except ValueError:
+                pass
+        if m.group(3):
+            try:
+                v = int(m.group(3))
+                typeargs.append(v)
+            except ValueError:
+                pass
```

Comment:
Not a huge fan of this section, its longer and more complicated than necessary. Here is a suggeseted refactor that should be functionally equivalent:

```python
for arg_group in (2, 3):
  try:
    typeargs.append(int(m.group(arg_group)))

  except (TypeError, ValueError):
   pass
```